subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -114,6 +126,8 @@ public void resolveBuildDependencies(ConfigurationInternal configuration, Resolv
     }
 
     public void resolveGraph(ConfigurationInternal configuration, ResolverResults results) {
```

Comment:
I couldn't quite get this to work. I am receiving a `java.lang.StackOverflowError`. Maybe this happening because we resolve the graph in a runnable build operation and then also try to get the resolution result via `configuration.getIncoming().getResolutionResult()`. The internal state of "being resolved" might not be set at this point of time. Does this sound like a reasonable assessment?

Reply:
You shouldn't need to do `configuration.incoming.resolutionResult`. Instead, try creating the lock file _after_ the delegate has performed the actual resolution, and use the result that has been produced. Something like:
```
@Override
public void resolveGraph(ConfigurationInternal configuration, ResolverResults results) throws ResolveException {
     delegate.resolveGraph(configuration, results);
     lockConfiguration(configuration, results.getResolutionResult());
}
```

Reply:
Thanks for the pointer on `ResolverResults`. I wasn't aware of the fact that you could get the `ResolutionResult` from there as well. That works.